maps-compose/src/main/java/com/google/maps/android/compose/CameraPositionState.kt

```diff
@@ -173,9 +173,13 @@ class CameraPositionState(
      * suspend until a map is bound and animation will begin.
      *
      * This method should only be called from a dispatcher bound to the map's UI thread.
+     *
+     * @param update the change that should be applied to the camera
+     * @param durationMillis the duration of the animation in milliseconds. The default animation
+     * duration is used if a negative value is provided.
      */
     @UiThread
-    suspend fun animate(update: CameraUpdate) {
+    suspend fun animate(update: CameraUpdate, durationMillis: Int = -1) {
```

Comment:
I don't have a strong opinion on this, but the variable name in the underlying Map API is `durationMs`:
https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap#animateCamera(com.google.android.gms.maps.CameraUpdate)

However, for the allowed values here, the underlying Maps API throws an `IllegalArgumentException` if this number is not positive:

>durationMs - The duration of the animation in milliseconds. This must be strictly positive, otherwise an IllegalArgumentException will be thrown.

Two questions/comments:
1. Should this API behavior match the underlying Maps SDK, in that `IllegalArgumentException` is thrown if the duration is not valid?
2. I don't believe `0` should be a valid value here based on the above documentation. Seems like if `0` is used, then the `animateCamera()` API shouldn't be called, and the `moveCamera()` API should be used instead.

Reply:
I deliberately did not want `IllegalArgumentException` to be propagated up since the semantics of this method is different (a negative value means the default duration will be used). However, `0` as an input is a good catch in which case calling `moveCamera` would be the right approach. 

On second thought, I'm leaning towards mirroring the SDK's behavior of throwing when `durationMs` is non-positive. We would need to overload `animate` since it won't be possible to set a default duration value without changing the semantics of the API (like I did in this change). This is also more consistent with other design decisions made in this library. Let me know if you have any thoughts here.